cni/pkg/util/podutil.go

```diff
@@ -104,10 +104,6 @@ func AnnotateEnrolledPod(client kubernetes.Interface, pod *metav1.ObjectMeta) er
 }
 
 func AnnotateUnenrollPod(client kubernetes.Interface, pod *metav1.ObjectMeta) error {
-	if pod.Annotations[annotation.AmbientRedirection.Name] != constants.AmbientRedirectionEnabled {
```

Comment:
Why do we drop this? I can see why it might be a good idea but its not obvious to me and doesn't seem explained in the description so want to make sure I grok it

Reply:
Yeah - I should have mentioned in the PR desc.

I removed this when debugging to rule out "unenroll silently not removing the label" because we happened to pass in the "wrong" (or an out of date) pod object, and left it out because I think it's more trouble than it's worth.

We don't do this for adding the annotation, and I don't want these basic helper funcs to make assumptions about staleness